modules/selection/main/index.coffee

```diff
@@ -271,7 +271,29 @@ class Selection
     this
 
   # clears the contents of a node and then adds the children passed in
-  set: (children) -> this.clear().add(children)
+  set: (children) ->
+    # The use of Promise.resolve can delay when the replacement happens, so
+    # check to only do this when needed. This makes testing nicer when adding
+    # non-promise content
+    if children.then
+      Promise.resolve(children).then((sel) => this.clear().add(sel))
```

Comment:
Should we use this promise logic for add/append chaining too? (i.e. the rest of the `Selection` module)